pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamConfig.java

```diff
@@ -142,7 +143,10 @@ public StreamConfig(String tableNameWithType, Map<String, String> streamConfigMa
     }
     _connectionTimeoutMillis = connectionTimeoutMillis;
 
-    int fetchTimeoutMillis = DEFAULT_STREAM_FETCH_TIMEOUT_MILLIS;
+    // For Kinesis, we need to set a higher fetch timeout to avoid getting stuck in empty records loop
+    int fetchTimeoutMillis =
+        !_consumerFactoryClassName.contains("KinesisConsumerFactory") ? DEFAULT_STREAM_FETCH_TIMEOUT_MILLIS_KINESIS
```

Comment:
Yeah, I can't use 10min as default because it will affect how soon the data is available in Pinot. We will just keep on waiting for kafka to return records